### PR TITLE
- Added saving to LS

### PR DESCRIPTION
**Pull request by NataliaHanzhuha**
- Added saving to LS
- Fixed default value of todos

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -3,27 +3,22 @@ import './App.css';
...
-      title: 'task3',
-      checked: false
-    },
-  ])
+  const TODOS_KEY = 'todos';
+  const [todos, setTodos] = useState(savedTodos())
   const [isNewItem, setNewItem] = useState(false)
   const [title, setTitle] = useState('');
 
+  function savedTodos() {
```
Comment (NikolasGumenyuk) [inline src/App.js:11]: you don't need it

File: src/App.js [hunk 2]
```diff
@@ -3,27 +3,22 @@ import './App.css';
...
-      title: 'task2',
-      checked: false
-    },
-    {
-      title: 'task3',
-      checked: false
-    },
-  ])
+  const TODOS_KEY = 'todos';
+  const [todos, setTodos] = useState(savedTodos())
```
Comment (NikolasGumenyuk) [inline src/App.js:7]: function as default state is not ok

**Review by NikolasGumenyuk — COMMENTED** [review]
It looks like a variant, but it is not true solution

File: src/App.js [hunk 3]
```diff
@@ -57,6 +52,11 @@ function App() {
     }
   }
 
+  function saveTodos(newTodos) {
+    localStorage.setItem(TODOS_KEY, JSON.stringify(newTodos));
+    setTodos(newTodos);
+  }
+
```
[the remark this answers is not in the document]
Comment (NataliaHanzhuha) [reply src/App.js]: used useEffect

File: src/App.js (hunk 1 above)
Comment (NataliaHanzhuha) [reply src/App.js]: removed

File: src/App.js (hunk 2 above)
Comment (NataliaHanzhuha) [reply src/App.js]: fixed

File: src/components/todo-list/TodoList.js [hunk 4]
```diff
@@ -0,0 +1,124 @@
+import { useState, useEffect } from 'react';
+import './TodoList.css';
+import { TodoItem } from '../todo-item/TodoItem';
+
+export function TodoList() {
+  const TODOS_KEY = 'todos';
+  const LSTodos = JSON.parse(localStorage.getItem(TODOS_KEY)) ?? [];
```
Comment (NikolasGumenyuk) [inline src/components/todo-list/TodoList.js:7]: use useEffect(with empty dependence) for it

File: src/components/todo-list/TodoList.js [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+          className='cancel-btn'
+          onClick={cancelNewItem}
+        >Cancel</button>
+      </div>
+      : <button
+        className="add-btn"
+        onClick={() => setNewItem(true)}>
+        Add Todo Item
+      </button>
+  }
```
Comment (NikolasGumenyuk) [inline src/components/todo-list/TodoList.js:98]: move it to new component

File: src/components/todo-list/TodoList.js [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+
+  return (
+    <div className='page'>
+      <header>
+        <h1>To Do List</h1>
+      </header>
+
+      {addNewTodoTemplate()}
```
Comment (NikolasGumenyuk) [inline src/components/todo-list/TodoList.js:106]: change it on new component like a ToDoListInput

File: src/components/todo-list/TodoList.js [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+    </ul>
+    return <>
+      <h2>{title}</h2>
+      {
+        filteredList?.length
+          ? list
+          : <p className='gray'>Emply list yet</p>
+      }
+    </>
+  }
```
Comment (NikolasGumenyuk) [inline src/components/todo-list/TodoList.js:72]: it must be a component

File: src/components/todo-list/TodoList.js [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+      <header>
+        <h1>To Do List</h1>
+      </header>
+
+      {addNewTodoTemplate()}
+
+      <div className='wrapper'>
+
+        <div className='list-wrapper'>
+          {todoList(false, 'In Progress')}
```
Comment (NikolasGumenyuk) [inline src/components/todo-list/TodoList.js:111]: it must be a component  ToDoList

File: src/components/todo-list/TodoList.js [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+        </div>
+
+        <div className='list-wrapper'>
+          {todoList(true, 'Done')}
```
Comment (NikolasGumenyuk) [inline src/components/todo-list/TodoList.js:115]: it must be a component ToDoList

**Review by NikolasGumenyuk — COMMENTED** [review]
its looks like more beautiful shit

File: src/components/todo-list/TodoList.js (hunk 4 above)
Comment (NataliaHanzhuha) [reply src/components/todo-list/TodoList.js:7]: fixed

File: src/components/todo-list/TodoList.js (hunk 5 above)
Comment (NataliaHanzhuha) [reply src/components/todo-list/TodoList.js:98]: moved

File: src/components/todo-list/TodoList.js (hunk 6 above)
Comment (NataliaHanzhuha) [reply src/components/todo-list/TodoList.js:106]: done

File: src/components/todo-list/TodoList.js (hunk 8 above)
Comment (NataliaHanzhuha) [reply src/components/todo-list/TodoList.js:111]: fixed

File: src/components/todo-list/TodoList.js (hunk 9 above)
Comment (NataliaHanzhuha) [reply src/components/todo-list/TodoList.js:115]: fixed

**Review by NikolasGumenyuk — APPROVED** [review]
this is what is needed, you are a big well done

**Pull request merged**